ctapipe/reco/hillas_intersection.py

```diff
@@ -218,6 +241,12 @@ def _predict(self, hillas_dict, array_pointing, telescopes_pointings=None):
         src_fov_lon, src_fov_lat, err_fov_lon, err_fov_lat = self.reconstruct_nominal(
             hillas_dict_mod
         )
+
+        # Catch events reconstructed at great angular distance from camera center
+        # and retrun INVALID container to prevent SkyCoord error below.
+        if far_outside_fov(src_fov_lat * u.rad, src_fov_lon * u.rad):
```

Comment:
It's relatively expensive to create new, single value quantities like this:
```
In [1]: import astropy.units as u

In [2]: %timeit 5 * u.deg
6.11 µs ± 285 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [3]: %timeit 5.0 * u.deg
5.33 µs ± 116 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [4]: %timeit u.Quantity(5.0, u.deg)
3.14 µs ± 194 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Together with the above comment, I'd not use astropy quantities here. You know these are in radians, so I'd implement the method above purely on floats. The constant could be defined like this:

```
FOV_ANGULAR_DISTANCE_LIMIT_RAD = (45 * u.deg).to_value(u.rad)
```